doc/ref_model/chapters/appendix-a.md

```diff
@@ -29,13 +29,21 @@ The goal is not to be prescriptive on how to re-architect existing or architect
 
-CNTT realizes there are implications to this and limitations to the ability to live by the Abstraction Principle. A textbook example of a Capability that transgresses this principle, is SR-IOV. Other, less notable, yet very important examples include vGPUs and other acceleration hardware, such as FPGA. A less obvious, yet critically important example is the VNF program(s) itself. As workloads execute native microarchitecture opcodes, those instruction sets effectively constitute an ABI (Application Binary Interface). Additionally, these programs may or may not, attempt to execute vendor-specific extensions to standard instruction sets, such as x86 or ARM.
+<a name="A.5.1"></a>
+### A.5.1 Feature Availability
+One such case is where CNTT strategically deems a given capability as mandatory, but the feature is a roadmap item, under development or otherwise unavailable. To address this scenario, a policy can be created to recognize the current state of the technology, and identify a Time Point (TP) in VNF Evolution when the feature will become mandatory for RC purposes. 
 
-Solving the problems associated with implementing the Abstraction Principle is a work in progress. CNTT has not solved all of the associated problems, nor has the industry. As technology evolves and more designs incorporate cloud native concepts, these problems will be addressed. This appendix is specifically intended to provide CNTT policies to manage these situations as they exist today,   and their exceptions and transitions, as the technology around and supporting the Abstraction Principle matures.
+<a name="A.5.2"></a>
+### A.5.2 Hardware-Dependent Coding
+Another such case is hardware-dependent coding. As described in the Principles sections of RM Chapter 1, features that require hardware-dependent code in a workload are prohibited in CNTT compliant implementations. This principle is henceforth referred to as the "Abstraction Principle". Note, this is not the case for Compute node host software (e.g., host OS). Within the Infra, hosts are expected to contain software customized for the specific hardware equipped. However, the intent is that these software drivers in conjunction with software at higher layers will abstract the capabilities they enable, thereby exposing them through an open, Cloud Native API. An example of exposing capabilities in this manner is implemented in the Virtio family of APIs. This requirement is in support of VNF abstraction and portability of VNFs across the Infra landscape.
```

Comment:
What is Cloud Native API?

Reply:
@pgoyal01 Hi Pankaj, Cloud Native API is merely descriptive; it's not intended to be a formal name, with a formal definition. In the very next sentence I explain it in more detail with a Virtio example, but if that doesn't clarify it enough, call me on Monday and I can explain it. Thanks, -Mark

Reply:
With respect to Cloud Native, is that what is meant here?  May I suggest 'open and neutral API'?

Reply:
@kedmison Hi Kelvin, The intent in that language is to explicitly drive people towards Cloud Native implementations, consuming/furnishing abstractable/abstracted virtual resources. I'm not sure neutral conveys that.

Reply:
I see.  I was thinking that we wanted to drive people a bit more towards APIs like virtio, etc which I thought of as neutral, but not really related to the cloud-native factors that are drivers for application design.    Perhaps its only me but your clarifying clause above "consuming/furnishing abstractable/abstracted virtual resources" I find a lot more illuminating.

Reply:
> 
> 
> I see. I was thinking that we wanted to drive people a bit more towards APIs like virtio

@kedmison That was the intent, the example I used, and I mentioned virtio explicitly by name. I don't follow how that is not driving them towards APIs like virtio. Thx, -Mark

Reply:
When I think Cloud Native APIs, virtio doesn't come to mind.  Perhaps it's just me.  Since you immediately follow the cloud-native reference with a virtio example, I'll resolve this conversation.